src/wallet/walletdb.cpp

```diff
@@ -540,30 +540,45 @@ static LoadResult LoadRecords(CWallet* pwallet, DatabaseBatch& batch, const std:
     return LoadRecords(pwallet, batch, key, prefix, load_func);
 }
 
+bool HasLegacyRecords(CWallet* pwallet)
+{
+    const auto& batch = pwallet->GetDatabase().MakeBatch();
+    return HasLegacyRecords(pwallet, *batch);
+}
+
+bool HasLegacyRecords(CWallet* pwallet, DatabaseBatch& batch)
+{
+    for (const auto& type : DBKeys::LEGACY_TYPES) {
+        DataStream key;
+        DataStream value{};
+        DataStream prefix;
+
+        prefix << type;
+        std::unique_ptr<DatabaseCursor> cursor = batch.GetNewPrefixCursor(prefix);
+        if (!cursor) {
+            // Could only happen on a closed db, which means there is an error in the code flow.
+            pwallet->WalletLogPrintf("Error getting database cursor for '%s' records", type);
```

Comment:
Just a question: Why log the same message that gets thrown?

Reply:
Because the caller is not logging it and didn’t want to change the existing behavior in the refactoring commit, the `pwallet` argument could have otherwise been removed.